### DISCUSSION TITLE
Re-use CLI parser?

### DISCUSSION THREAD

**Discussion by TobiX** [python-gitlab/python-gitlab#2882] [Ideas]
I find myself writing a lot of (reusable) one-of scripts, so it would be really nice to have all the "standard" options from the command line tool. Most of the time I use `Gitlab.from_config()`, but that requires users of my script to create a config file to use them. What I would like is something like `GitLab.from_environment()` or something, which would create a default argument parser so I could use all options of the `gitlab` tool for my own tools. Maybe even provide something like
```python
parser = argparse.ArgumentParser(description="My GitLab API Tool", allow_abbrev=False)
GitLab.add_arguments(parser)
args = parser.parse_args()
GitLab.from_environment(args=args)
```
which would allow to integrate all the "config" options from the command line tool into other tools.

How about it? (I might have some time to contribute the change to facilitate this, probably requires a bit of refactoring...)

(This seems a bit similar to #2830)

**Comment by nejch** [comment]
We could probably extract some of the config logic out of the CLI's `main()` function and then use that directly in the `Gitlab.merge_config()` class method.

I'm personally a bit busy at the moment with other things so contributions are welcome! Will be happy to review MRs.
